caches.go

```diff
@@ -304,6 +303,8 @@ func (krc *keyRegionCache) put(reg hrpc.RegionInfo) (overlaps []hrpc.RegionInfo,
 	}
 	krc.m.Unlock()
 
+	// -(N overlapped regions) + (1 new region)
+	cachedRegionTotal.Add(-float64(len(overlaps)) + 1)
```

Comment:
I think what you have here is correct, but it's a little tricky. Something I would have more confidence in is if you were to get the length before and after modification and add the diff to the counter. ie.

```
krc.m.Lock()
beforeLen := krc.regions.Len()
krc.regions.Put(...
...
for _, o := range overlaps {
    ...
}
afterLen := krc.regions.Len()
krc.m.Unlock()

cachedRegionTotal.Add(float64(afterLen-beforeLen))
```
Up to you, whichever you prefer.

Reply:
Good suggestion!